### PR TITLE
Remove stale useless default_endpoint field in gateway.server

### PR DESCRIPTION
**Pull request by hzxuzhonghu**

### CODE REVIEW

**Comment by geeknoid** [comment]
This is a breaking change. Do we need a deprecation story here?

**Comment by linsun** [comment]
How do we know this is stale?  we definitely need release note this deprecation.

File: networking/v1alpha3/gateway.proto [hunk 1]
```diff
@@ -410,10 +410,8 @@ message Server {
   // https, and the TLS modes to use.
   TLSOptions tls = 3;
 
-  // The loopback IP endpoint or Unix domain socket to which traffic should
-  // be forwarded to by default. Format should be `127.0.0.1:PORT` or
-  // `unix:///path/to/socket` or `unix://@foobar` (Linux abstract namespace).
-  string default_endpoint = 5;
+  reserved 5;
+  reserved "default_endpoint";
```
Comment (rshriram) [inline networking/v1alpha3/gateway.proto]: why is this stale? this is implemented right? if not, my bad as I forgot about implementing this. This needs to be implemented.

Comment (hzxuzhonghu) [reply networking/v1alpha3/gateway.proto]: It is no implemented now.

**Comment by hzxuzhonghu** [comment]
@linsun @geeknoid I noticed it was in since 2017, and not used

**Comment by rshriram** [comment]
I think you are looking at a wrong branch.. This field was added 5 months ago by me
https://github.com/istio/api/pull/742 
I didn't implement it (forgot to)

**Comment by hzxuzhonghu** [comment]
ok, then let me close it.

**Pull request closed without merging**